api/tests/Feature/UserRoleTest.php

```diff
@@ -249,8 +250,8 @@ public function testAdminCanAddAndRemoveNonTeamRoleToUser()
     // Create a user with an old role.  Assert that the admin can remove the old role and add the new role, now with teams!
     public function testAdminCanAddAndRemoveTeamRoleToUser()
     {
-        $oldRole = Role::factory()->create(['is_team_based' => true]);
-        $newRole = Role::factory()->create(['is_team_based' => true]);
+        $oldRole = Role::where('name', 'pool_operator')->sole();
+        $newRole = Role::where('name', 'process_operator')->sole();
```

Comment:
Was there anything wrong with how this worked before? I'm just thinking about how we'll soon want to deprecate and remove `pool_operator`.

Reply:
Yeah now the user policy is explicit about what roles it allows to be added, unknown roles are now rejected.

Reply:
Switched to use the new role only in https://github.com/GCTC-NTGC/gc-digital-talent/pull/10791/commits/9768145979dfd45cf300dfdf9249990efe053c15